test/extensions/transport_sockets/http_11_proxy/connect_test.cc

```diff
@@ -386,25 +382,32 @@ TEST_F(SocketFactoryTest, CreateSocketReturnsNullWhenInnerFactoryReturnsNull) {
 }
 
 TEST(ParseTest, TestValidResponse) {
+  unsigned long len;
```

Comment:
Do we have test coverage of the incomplete case?
Can we add tests for all the termination variants? \r\n\r\n \r\n\n \n\r\n \n\n ?
It might also be a good idea to add a test in which there are response headers in addition to the response firstline, just to make sure we've got this nailed.